### PR TITLE
Accept provision/delete calls from PVCs that have the 'migrated-to' annotation of the provisioner name

### PR DESCRIPTION
**Pull request by davidz627**
/kind feature

Final piece to fix: https://github.com/kubernetes/kubernetes/issues/79043

```release-note
CSI Migration: Fixes CSI migration issue where PVCs provisioned before migration was turned on could not be deleted
```

### CODE REVIEW

**Comment by davidz627** [comment]
/cc @msau42 @jsafrane 
this PR is ready for review

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -403,12 +404,16 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 		return nil, controller.ProvisioningFinished, errors.New("storage class was nil")
 	}
 
-	if options.PVC.Annotations[annStorageProvisioner] != p.driverName {
+	if options.PVC.Annotations[annStorageProvisioner] != p.driverName && options.PVC.Annotations[annMigratedTo] != p.driverName {
```
Comment (msau42) [inline pkg/controller/controller.go]: do we have a unit test for this?

Comment (davidz627) [reply pkg/controller/controller.go]: now we do :)

**Comment by msau42** [comment]
/lgtm
/approve

Can you make the release note more easily understandable by someone that doesn't have background on the issue? Something like "Fixes CSI migration issue where PVCs provisioned before migration was turned on could not be deleted"
/hold

**Pull request merged**